examples/simple-usage/lifecycle.ts

```diff
@@ -0,0 +1,23 @@
+import { TaskExecutor } from "@golem-sdk/golem-js";
+
+/**
+ * This example shows how to use the TaskExecutor lifecycle events.
+ */
+(async function main() {
+  console.log("Creating TaskExecutor and registering for lifecycle events...");
+  const executor = await TaskExecutor.create("golem/alpine:latest");
+
+  executor.events.on("terminating", () => {
+    console.log("Lifecycle: TaskExecutor is about to be terminated");
+  });
+
+  executor.events.on("terminated", () => {
+    console.log("Lifecycle: TaskExecutor is terminated. Exiting...");
+    process.exit(0);
+  });
+
+  console.log("Calling TaskExecutor.end()");
+  await executor.end();
+  // process.exit() is called in an event handler above, so this message should not be displayed.
+  console.log("Should not display this message.");
```

Comment:
Pretty cool example with `process.exit` in the callback, but IMO we should not promote such practices

Reply:
Actually, why not?

Reply:
In my opinion, using `process.exit` is in general a bad practice (because it prevents node from cleaning up async/ IO operations) and I avoid it unless I have a very good reason to use it. 

Also this particular example - write to stdout and exit immediately - is exactly what node docs advise *not* to do: https://nodejs.org/api/process.html#processexitcode

Reply:
I support this argument. `process.exit` is frequently miss-used by developers and the hinted usage here is not a good one. In general, I think that this example is really serving a purpose of a unit test. You have a good unit test here already so there's no need to create that example at all.